csrc/multidevice/utils.cpp

```diff
@@ -555,12 +555,12 @@ std::set<DeviceIdxType> involvedDevices(Expr* expr) {
        {ir_utils::filterByType<TensorView>(expr->inputs()),
         ir_utils::filterByType<TensorView>(expr->outputs())}) {
     for (auto* tv : tvs) {
-      NVF_ERROR(
-          tv->hasDeviceMesh(),
-          "the TensorView has no device mesh: ",
-          tv->toString());
-      auto& mesh = tv->getDeviceMesh().vector();
-      std::copy(mesh.begin(), mesh.end(), std::inserter(ret, ret.end()));
+      if (tv->hasDeviceMesh()) {
+        auto& mesh = tv->getDeviceMesh().vector();
+        std::copy(mesh.begin(), mesh.end(), std::inserter(ret, ret.end()));
+      } else {
+        ret.insert(0);
```

Comment:
Why is this change needed?

Reply:
`involvedDevices` is now called from `HostIrEvaluator` which can take-in single device fusion where TVs don't have device mesh. In a single device Fusion, it is natural that this function returns `device 0`